modules/loader/src/main/java/com/opengamma/strata/loader/csv/FullSwapTradeCsvLoader.java

```diff
@@ -147,27 +152,106 @@
+    List<FloatingRateIndex> indices = new ArrayList<>();
+    Set<DayCount> dayCounts = new LinkedHashSet<>();
+    boolean missingDayCount = false;
+    String legPrefix = "Leg 1 ";
+    Optional<String> payReceiveOpt = Optional.of(getValue(row, legPrefix, DIRECTION_FIELD));
     int i = 1;
-    while (payReceive.isPresent()) {
-      legs.add(parseLeg(row, "Leg " + i + " "));
+    while (payReceiveOpt.isPresent()) {
+      // parse this leg
+      FloatingRateIndex index = parseIndex(row, legPrefix);
+      if (index != null) {
+        // floating leg
+        indices.add(index);
+        dayCounts.add(index.getDefaultFixedLegDayCount());
+      } else {
+        // fixed leg
+        indices.add(null);
+        if (!findValue(row, legPrefix, DAY_COUNT_FIELD).isPresent()) {
+          missingDayCount = true;
+        }
+      }
+      // check if there is another leg
       i++;
-      payReceive = findValue(row, "Leg " + i + " ", DIRECTION_FIELD);
+      legPrefix = "Leg " + i + " ";
+      payReceiveOpt = findValue(row, legPrefix, DIRECTION_FIELD);
     }
+    // determine the default day count for the fixed leg
+    DayCount defaultFixedLegDayCount = null;
+    if (missingDayCount) {
+      if (dayCounts.size() > 1) {
+        throw new IllegalArgumentException(
+            "Swap leg must define day count on fixed legs when more than 2 floating legs");
+      }
+      defaultFixedLegDayCount = dayCounts.iterator().next();
+    }
+    // parse fully now we know the number of legs and the default fixed leg day count
+    List<SwapLeg> legs = parseLegs(row, indices, defaultFixedLegDayCount);
     Swap swap = Swap.of(legs);
     return SwapTrade.of(info, swap);
   }
 
+  //-------------------------------------------------------------------------
+  // parse the index and default fixed leg day count
+  private static FloatingRateIndex parseIndex(CsvRow row, String leg) {
+    Optional<String> fixedRateOpt = findValue(row, leg, FIXED_RATE_FIELD);
+    Optional<String> indexOpt = findValue(row, leg, INDEX_FIELD);
+    if (fixedRateOpt.isPresent()) {
+      if (indexOpt.isPresent()) {
+        throw new IllegalArgumentException(
+            "Swap leg must not define both '" + leg + FIXED_RATE_FIELD + "' and  '" + leg + INDEX_FIELD + "'");
+      }
+      return null;
+    }
+    if (!indexOpt.isPresent()) {
+      throw new IllegalArgumentException(
+          "Swap leg must define either '" + leg + FIXED_RATE_FIELD + "' or  '" + leg + INDEX_FIELD + "'");
+    }
+    // use FloatingRateName to identify Ibor vs other
+    String indexStr = indexOpt.get();
+    FloatingRateName frn = FloatingRateName.parse(indexStr);
+    if (frn.getType() == FloatingRateType.IBOR) {
+      // re-parse Ibor using tenor, which ensures tenor picked up from indexStr if present
+      Frequency freq = Frequency.parse(getValue(row, leg, FREQUENCY_FIELD));
+      Tenor iborTenor = freq.isTerm() ? frn.getDefaultTenor() : Tenor.of(freq.getPeriod());
```

Comment:
Not immediately related to this PR but our getDefaultTenor() logic is not fully correct here.

Default tenor for most major currencies is 6M, with only USD being 3M

Reply:
```
* This is useful for providing a basic default where errors need to be avoided.
* The value returned is not intended to be based on market conventions."
```